src/test/java/org/springframework/integration/kafka/dsl/KafkaDslTests.java

```diff
@@ -178,15 +181,18 @@ public IntegrationFlow sendToKafkaFlow() {
 			return f -> f
 					.<String>split(p -> Stream.generate(() -> p).limit(101).iterator(), null)
 					.publishSubscribeChannel(c -> c
-							.subscribe(sf -> sf.handle(kafkaMessageHandler(producerFactory(), TEST_TOPIC),
-									e -> e.id("kafkaProducer")))
-							.subscribe(sf -> sf.handle(kafkaMessageHandler(producerFactory(), TEST_TOPIC3),
-									e -> e.id("kafkaProducer3")))
+							.subscribe(sf -> sf.handle(
+									kafkaMessageHandler(producerFactory(), TEST_TOPIC1)
+											.timestampExpression("T(Long).valueOf('1487694048633')"),
+									e -> e.id("kafkaProducer1")))
+							.subscribe(sf -> sf.handle(kafkaMessageHandler(producerFactory(), TEST_TOPIC2)
+											.timestampExpression("T(System).currentTimeMillis()"),
```

Comment:
When you have a `Function` variant, this will be pretty straightforward for Java codding:
```
.timestampExpression(m -> System.currentTimeMillis())
```